### PR TITLE
Authentication Implementation

### PR DESCRIPTION
**Pull request by hellen-22**
# Description

Implemented authentication and implemented creation of accounts.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## How Has This Been Tested?
Wrote tests for the various endpoints.

- [x] Conftest
- [x] Account Serializer
- [x] User Serializer
- [x] Models
- [x] Add CI pytest

**Test Configuration**:
* Firmware version:
* Hardware:
* Toolchain:
* SDK:

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by JimmyTron — COMMENTED** [review]
So far its ok check on highlighted areas if you need to make changes

**Review by JimmyTron — APPROVED** [review]
Satisfies the basics and covers the issue of user creation and  accounts. 
since we will have one instance of the user but various instances of which our users would be able to interact.

File: .github/workflows/django.yml [hunk 1]
```diff
@@ -29,3 +29,7 @@ jobs:
     - name: Run Tests
       run: |
         python manage.py test
+
+    - name: Run Pytest
+      run: |
+        pytest
```
[the remark this answers is not in the document]
Comment (OyugoObonyo) [reply .github/workflows/django.yml:35]: Try using the Django unit-tests already adopted by the project. One of the jobs must go, it doesn't make sense to have 2 jobs running the same exact tests.

File: accounts/serializers.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
+from rest_framework import serializers
+from django.db import transaction
+
+from accounts.models import User, Account
+
+"""Serializer to display account details to be used in the UserSerializer"""
+class AccountDetailSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Account
+        fields = ['id', 'account_name', 'bio']
```
[the remark this answers is not in the document]
Comment (OyugoObonyo) [reply accounts/serializers.py:10]: Attempt resolving the display photo 301 error.

File: accounts/views.py [hunk 3]
```diff
@@ -1,3 +1,38 @@
...
+    permission_classes = [IsAdminUser]
+
+    def get_queryset(self):
+        if self.request.method == 'GET':
+            return User.objects.annotate(number_of_accounts=Count('account'))#Adds a new column that counts the number of accounts.
+        return User.objects.all()
+
+
+"""Api view to be used when a user first registers to the system"""
+class RegisterAccountViewSet(CreateModelMixin, RetrieveModelMixin, ListModelMixin, GenericViewSet):
```
Comment (OyugoObonyo) [inline accounts/views.py:23]: Preferably, create  Retrieve view set separately from the Register Account View set. A retrieve or list view isn't required at a Register account viewset

File: accounts/views.py [hunk 4]
```diff
@@ -1,3 +1,38 @@
...
+    queryset = Account.objects.all()
+    serializer_class = UserAccountRegistrationSerializer
+    permission_classes = [IsAdminUser]
```
Comment (OyugoObonyo) [inline accounts/views.py:26]: Admin permissions to register an account isn't really ideal. Reconsider this.

**Review by OyugoObonyo — CHANGES_REQUESTED** [review]
Please consider making the suggested changes. Revert back in case of any issues.

File: accounts/serializers.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (hellen-22) [reply accounts/serializers.py:10]: > What of profile pic. I suppose the account could have more fields

Profile picture is an optional field. Do i really need to add it to the fields.

[the remark this answers is not in the document]
Comment (hellen-22) [reply accounts/serializers.py:10]: > Attempt resolving the display photo 301 error.

I figured out what the issue was. I'd just added it to the fields and not in the creation of model step.

**Pull request merged**